streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -613,10 +604,14 @@ private long sumOfChangelogOffsets(final TaskId id, final Map<TopicPartition, Lo
         for (final Map.Entry<TopicPartition, Long> changelogEntry : changelogOffsets.entrySet()) {
             final long offset = changelogEntry.getValue();
 
-            offsetSum += offset;
-            if (offsetSum < 0) {
-                log.warn("Sum of changelog offsets for task {} overflowed, pinning to Long.MAX_VALUE", id);
-                return Long.MAX_VALUE;
+            if (offset == Task.LATEST_OFFSET) {
+                return Task.LATEST_OFFSET;
+            } else {
+                offsetSum += offset;
```

Comment:
It might be nice to have a sanity check here that `offset` is non-negative, since that would indicate we've unexpectedly received a sentinel value. I thought we did that already, but it's obviously not here.

Reply:
We put the check for negative offsets in the RecordCollector, but I guess it doesn't hurt to check again here?

Reply:
Aha! Thanks. Yeah, I'd be in favor of coding defensively here as well.

Reply:
Ack